File: 0150-evaluate-reverse-polish-notation/0150-evaluate-reverse-polish-notation.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        stack = []
        for s in tokens:
            if s in '+-*/':
                b = stack.pop()
                a = stack.pop()
                if s == '+':
                    stack.append(a + b)
                elif s == '-':
                    stack.append(a - b)
                elif s == '*':
                    stack.append(a * b)
                else:
                    stack.append(int(a / b))
            else:
                stack.append(int(s))
        return stack.pop()

File: 0150-evaluate-reverse-polish-notation/0150-evaluate-reverse-polish-notation.cpp
class Solution
{
public:
    bool isdigitt(string x)
    {
        if(isdigit(x[0]))
            return true;
        else if(x[0]=='-' and x.length()>1)
            return true;
        return false;
    }
    int evalRPN(vector<string> &tokens)
    {
        stack<int> s;
        for (string x : tokens)
        {
            if (isdigitt(x))
                s.push(stoi(x));
            else
            {
                int val2 = s.top();
                s.pop();
                int val1 = s.top();
                s.pop();
                if (x == "+")
                    s.push(val1 + val2);
                else if (x == "-")
                    s.push(val1 - val2);
                else if (x == "*")
                    s.push(val1 * val2);
                else if (x == "/")
                    s.push(val1 / val2);
            }
        }
        return s.top();
    }
};